provider/ec2/environ.go

```diff
@@ -1592,17 +1614,16 @@ func (e *environ) allInstanceIDs(ctx context.ProviderCallContext, filter *ec2.Fi
 	return ids, nil
 }
 
-func (e *environ) allInstances(ctx context.ProviderCallContext, filter *ec2.Filter) ([]instances.Instance, error) {
-	resp, err := e.ec2.Instances(nil, filter)
+func (e *environ) allInstances(ctx context.ProviderCallContext, filters []*sdkec2.Filter) ([]instances.Instance, error) {
+	req := &sdkec2.DescribeInstancesInput{Filters: filters}
+	resp, err := e.ec2Client.DescribeInstances(req)
 	if err != nil {
 		return nil, errors.Annotate(maybeConvertCredentialError(err, ctx), "listing instances")
 	}
 	var insts []instances.Instance
 	for _, r := range resp.Reservations {
-		for i := range r.Instances {
-			inst := r.Instances[i]
-			// TODO(wallyworld): lookup the details to fill in the instance type data
```

Comment:
I'm not sure what this comment meant, but seems like we don't need it anymore?

Reply:
Comment was from 2013!!! 
We used to have a Metadata() method on instances.
There's 2 of these comments in the ec2 package, they can be deleted